pkg/api/filters/headers/validator.go

```diff
@@ -0,0 +1,56 @@
+package headers
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Azure/open-service-broker-azure/pkg/api/filters"
+	log "github.com/Sirupsen/logrus"
+)
+
+var responseMissingAPIVersion = []byte(
+	`{ "error": "MissingAPIVersion", "description": "The request did not ` +
+		`include the X-Broker-API-Version header"}`,
+)
+
+var responseAPIVersionIncorrect = []byte(
+	`{ "error": "APIVersionIncorrect", "description": "X-Broker-API-Verson ` +
+		`header includes an incompatible version"}`,
+)
+
+type validator struct{}
+
+//NewValidator creates a new instance of the header validator
+func NewValidator() filters.Filter {
+	return &validator{}
+}
+
+//Filter validates that required headers are present
+func (v *validator) Execute(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		apiVersion := r.Header.Get("X-Broker-API-Version")
+		if apiVersion == "" {
+			v.sendError(w, responseMissingAPIVersion)
+			return
+		}
+		if !strings.HasPrefix(apiVersion, "2.") {
+			v.sendError(w, responseAPIVersionIncorrect)
+			return
+		}
+		//call the original handler.
+		handler(w, r)
+	}
+}
+
+func (v *validator) sendError(
+	w http.ResponseWriter,
+	responseBody []byte,
+) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusPreconditionFailed)
+	if _, err := w.Write(responseBody); err != nil {
+		log.WithField("error", err).Error(
```

Comment:
Bad request != error from the broker's perspective. (Litmus test is whether an operator would want to know about it.) Can we downgrade this to info or debug?

Reply:
👍

Reply:
Actually, in this case it's a failure writing the response, not the filter failure itself. I think this one should still be Error no? In this case, the _broker_ failed to do something. That log only happens when err != nil on the write.

Reply:
You're right. I misread.